hydra-node/test/Hydra/Chain/Direct/TxSpec.hs

```diff
@@ -140,6 +145,28 @@ spec =
                         & counterexample "Failed to construct and observe init tx."
                         & counterexample (renderTx tx)
                         & counterexample (show e)
+    describe "commitTx" $ do
+      prop "Appends needed commit changes to blueprintTx" $ \blueprintTx' ->
+        forAll genScriptRegistry $ \scriptRegistry ->
+          forAll arbitrary $ \party ->
+            forAll (genAbortableOutputs [party] `suchThat` (\(a, _) -> not (null a))) $ \(initials, commits) ->
+              forAll arbitrary $ \vkh -> do
+                initialInputAndOutput <- elements initials
+                let utxoToCommit = foldMap third commits
```

Comment:
This is wrong. `genAbortableOutputs` produces initial and commit outputs. The latter are outputs of commit transactions (!). You don't want to use them here. Instead, this utxo should (at least) contain the inputs of the `blueprintTx'` which you pass to `commitTx`.